### ISSUE TITLE
API should accept RFC7235 authorization header

### ISSUE THREAD

**Issue by viceice** [hexpm/specifications#34]
Currently HEX API requires `authorization: <token>` as api header, it should support normal `Bearer` token as other API's will do.

eg
```bash
curl --include \
     --header "Accept: application/json" \
     --header "Authorization: Bearer <token>" \
  'https://private-anon-d3fb121a9d-hexpm.apiary-mock.com/packages?sort=&search='
```

- renovatebot/renovate#11028
- renovatebot/renovate#11168
- https://datatracker.ietf.org/doc/html/rfc7235#appendix-C

**Comment by ericmj** [comment]
Agreed. We need to implement it in a backwards compatible way on https://github.com/hexpm/hexpm first then the specifications can be updated.
